src/telemetry.c

```diff
@@ -1016,6 +1090,58 @@ int tm_set_payload(struct telem_ref *t_ref, char *payload)
         return ret;
 }
 
+
+/**
+ * Checks for id to have length = 32 characters
+ * and hexadecimal characters only.
+ *
+ * @param id A pointer to string to be checked
+ *
+ * @return 0 on success, 1 on failure
+ *
+ */
+static int validate_event_id(char *id)
+{
+       int rc = 0;
+       const char alphab[] = EVENT_ID_ALPHAB;
+
+       // 32 - 32 = 0
+       if((rc = (int)(strlen(id) - EVENT_ID_LEN)) != 0) {
+               return 1;
+       }
+
+       // verify alphabet
+       // strspn checks for characters in alphab
+       if(id[strspn(id, alphab)] != 0) {
```

Comment:
Since you already calculated strlen(id), you could just check that this is equal to the return value of strspn.